### PR TITLE
Add Timeout setting to tests

### PR DESCRIPTION
**Pull request by slozier**

### CODE REVIEW

File: Src/IronPythonTest/Cases/CaseExecuter.cs [hunk 1]
```diff
@@ -169,19 +169,38 @@ class CaseExecuter {
...
-                if(testcase.Options.Redirect && exitCode != 0) {
-                    Console.Error.WriteLine($"Error running {proc.StartInfo.FileName} {proc.StartInfo.Arguments}: ");
-                    Console.Error.WriteLine(output);
-                    Console.Error.WriteLine();
+
+                if (!proc.WaitForExit(testcase.Options.Timeout)) {
+                    proc.Kill();
+                    Console.Error.Write($"Timed out after {testcase.Options.Timeout / 1000.0} seconds.");
                 }
+                exitCode = proc.ExitCode;
```
Comment (slide) [inline Src/IronPythonTest/Cases/CaseExecuter.cs]: What will proc.ExitCode be if it's killed?

File: Src/IronPythonTest/Cases/AllCPythonCasesManifest.ini [hunk 2]
```diff
@@ -2,7 +2,7 @@
 Ignore=false
 Arguments="$(TEST_FILE)"
 WorkingDirectory=$(ROOT)/Src/StdLib/Lib
-Redirect=false
+Redirect=true
```
Comment (slide) [inline Src/IronPythonTest/Cases/AllCPythonCasesManifest.ini]: You wanted to redirect for all tests?

Comment (slide) [reply Src/IronPythonTest/Cases/AllCPythonCasesManifest.ini]: Disregard, I missed the only running test_subprocess in this for debugging.

File: Src/IronPythonTest/Cases/CaseExecuter.cs (hunk 1 above)
Comment (slozier) [reply Src/IronPythonTest/Cases/CaseExecuter.cs]: Looks like it's -1.

File: Src/IronPythonTest/Cases/AllCPythonCasesManifest.ini (hunk 2 above)
Comment (slozier) [reply Src/IronPythonTest/Cases/AllCPythonCasesManifest.ini]: As long as it's not causing issues with a specific test then I think we should.

**Comment by slozier** [comment]
test_subprocess timed out on `test_stderr_redirect_with_no_stdout_redirect`. Close and reopen to try again.

**Comment by slozier** [comment]
No timeout on the second attempt. Close and reopen to try again.

**Pull request merged**